File: client/src/app/models/product.ts
// 2 ways that are almost the same for modeling - type and interface

// INTERFACE
// export interface Product {
//   id: number;
//   name: string;
//   description: string;
//   price: number;
//   pictureUrl: string;
//   type: string;
//   brand: string;
//   quantityInStock: number;
// }

// TYPE
export type Product = {
  id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  type: string;
  brand: string;
  quantityInStock: number;
};

File: API/Entities/Product.cs
namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public long Price { get; set; } // Using long to match the price in Stripe. Otherwise we would use decimal for fraction
        public required string PictureUrl { get; set; }
        public required string Type { get; set; }
        public required string Brand { get; set; }
        public int QuantityInStock { get; set; }
        public string? PublicId { get; set; }

    }
}
